subsys/bluetooth/services/fast_pair/fp_keys.c

```diff
@@ -256,7 +301,7 @@ int fp_keys_generate_key(const struct bt_conn *conn, struct fp_keys_keygen_param
 		}
 	} else {
 		key_gen_failure_cnt = 0;
-		if (proc->state == FP_STATE_USE_TEMP_KEY) {
+		if (is_key_generated(proc->state)) {
```

Comment:
I think we could even assert that now with current implementation (the key must be generated after successful key generation ;) )